### ISSUE TITLE
Wiki Search not working as expected

### ISSUE THREAD

**Issue by upccjesse** [frappe/wiki#188]
```
erpnext 14.44.1
frappe 14.52.1
wiki 2.0.0
```

Wiki pages are published, but not guest accessible.

* All Wiki page routes are in the form of `wiki/page1`
* All Wiki pages are added to a Wiki Space with route `wiki`
* In Wiki Settings, the default space is the `wiki` space

When navigating to a Wiki page, i.e. `example.com/wiki/page1`, I have the Search Bar present. I can click it and execute a search, but generally I get no results.
* If I search for `Troubleshooting`, which I know that there are several pages with with `Troubleshooting` both in the title and the content of the page, I get no results. (see example 1)
* If I search for `wiki`, I get a random assortment of Wiki Page results that I can click on and navigate to, but the results are literally just the route of the page - the title and content are not present in the search results. (see example 2)

### Expected Behavior

I expect the search function to show relevant search results for terms that I know are present in the Wiki.

### Example 1: 
![image](https://github.com/frappe/wiki/assets/31744530/d1cddf71-d4dc-42fe-bf38-89d5c502112a)
![image](https://github.com/frappe/wiki/assets/31744530/a5280f03-5316-4964-85e4-ab05bc7215cf)
![image](https://github.com/frappe/wiki/assets/31744530/7f9f1f5a-972e-4e78-b013-831c64a8c0c4)

### Example 2:
![image](https://github.com/frappe/wiki/assets/31744530/ecdb9601-d997-41e2-a6d5-66427f84b561)
![image](https://github.com/frappe/wiki/assets/31744530/1e2e2d51-acb9-4003-aafa-59d304a56fa3)
![image](https://github.com/frappe/wiki/assets/31744530/2810d701-a9ec-4789-b114-0e4f37c765d7)

### Other info

I noticed that `Wiki Page` was not in the Global Search Settings list, so I added it, but nothing changed.

As far as I can tell, the search path once submitted to the server goes as such:

1. The HTTP Query contains parameter `cmd` of `wiki.wiki.doctype.wiki_page.search.search` which translates to the [`search` function located in `wiki/wiki/doctype/wiki_page/search.py`](https://github.com/frappe/wiki/blob/b70b44cf91e0b571142479667b1c9cdc724e0f1e/wiki/wiki/doctype/wiki_page/search.py#L18)
2. Since I don't have redis enabled in Wiki Settings, it defaults to the Frappe web search. This function calls `web_search` which is `from frappe.search import web_search`
3. This function is defined in the `frappe/frappe` Git Repository [here](https://github.com/frappe/frappe/blob/135fb6a486ac08ed5264038ba7dc77f04d21b8ed/frappe/utils/global_search.py#L494-L510)
4. This instantiates a new `WebsiteSearch` object with `index_name="web_routes"` as its sole parameter, then calls its `search` function. `WebsiteSearch` is a wrapper on top of `FullTextSearch` class, which is where the `search` function [is defined](https://github.com/frappe/frappe/blob/135fb6a486ac08ed5264038ba7dc77f04d21b8ed/frappe/search/full_text_search.py#L110).
5. This calls a `Whoosh` searcher to do the actual search. I haven't traced any further than this.

**Comment by BreadGenie** [comment]
Can you try enabling redis from Wiki Settings and try again?

**Comment by upccjesse** [comment]
I tried redis already, no change.
`rebuild-search-index` doesn't appear to be a command, but `build-search-index` and `rebuild-global-search` are.

`rebuild-global-search` did not have a noticeable effect.
`build-search-index`, however, did. I am getting the search results I expected.

2 questions, though:
1. Does the search index need to be built/rebuilt after every new wiki page is added?
2. Should this run automatically and, if so, how do I make sure it's running automatically?

**Comment by BreadGenie** [comment]
>     1. Does the search index need to be built/rebuilt after every new wiki page is added?

No, the the index for a new wiki page is built whenever it is created. Not sure what exactly is impeding from rebuilding the index.

**Comment by upccjesse** [comment]
There was no difference when redis was enabled, aside from the HTTP response on search indicating that redis was enabled.

**Comment by federicocalvo** [comment]
Hi, in my case any of this command works for mi. I have a wiki in docker with the last changes.
